File: rhinocommon/snippets/from_rhinocommon/vbnet/ex_dividebysegments.vb
Imports Rhino
Imports Rhino.DocObjects
Imports Rhino.Commands
Imports Rhino.Input
Imports Rhino.Geometry

Namespace examples_vb
  Public Class DivideBySegmentsCommand
    Inherits Rhino.Commands.Command
    Public Overrides ReadOnly Property EnglishName() As String
      Get
        Return "vbDivideCurveBySegments"
      End Get
    End Property

    Protected Overrides Function RunCommand(doc As RhinoDoc, mode As RunMode) As Result
      Const filter As ObjectType = ObjectType.Curve
      Dim objref As ObjRef = Nothing
      Dim rc = RhinoGet.GetOneObject("Select curve to divide", False, filter, objref)
      If rc <> Result.Success OrElse objref Is Nothing Then
        Return rc
      End If

      Dim curve = objref.Curve()
      If curve Is Nothing OrElse curve.IsShort(RhinoMath.ZeroTolerance) Then
        Return Result.Failure
      End If

      Dim segment_count = 2
      rc = RhinoGet.GetInteger("Divide curve into how many segments?", False, segment_count)
      If rc <> Result.Success Then
        Return rc
      End If

      Dim points As Point3d() = Nothing
      curve.DivideByCount(segment_count, True, points)
      If points Is Nothing Then
        Return Result.Failure
      End If

      For Each point As Point3d In points
        doc.Objects.AddPoint(point)
      Next

      doc.Views.Redraw()
      Return Result.Success
    End Function
  End Class
End Namespace

File: rhinocommon/snippets/from_rhinocommon/cs/ex_dividebysegments.cs
using Rhino;
using Rhino.DocObjects;
using Rhino.Commands;
using Rhino.Input;
using Rhino.Geometry;

namespace examples_cs
{
  public class DivideBySegmentsCommand : Rhino.Commands.Command
  {
    public override string EnglishName
    {
      get { return "csDivideCurveBySegments"; }
    }

    protected override Result RunCommand(RhinoDoc doc, RunMode mode)
    {
      const ObjectType filter = ObjectType.Curve; 
      ObjRef objref;
      var rc = RhinoGet.GetOneObject("Select curve to divide", false, filter, out objref);
      if (rc != Result.Success || objref == null)
        return rc;
  
      var curve = objref.Curve();
      if (curve == null || curve.IsShort(RhinoMath.ZeroTolerance))
        return Result.Failure;

      var segment_count = 2;
      rc = RhinoGet.GetInteger("Divide curve into how many segments?", false, ref segment_count);
      if (rc != Result.Success)
        return rc;

      Point3d[] points;
      curve.DivideByCount(segment_count, true, out points);
      if (points == null)
        return Result.Failure;
  
      foreach (var point in points)
        doc.Objects.AddPoint(point);
  
      doc.Views.Redraw();
      return Result.Success;
    }
  }
}